lib/worker/lib/workerLib.ts

```diff
@@ -231,10 +231,16 @@ export class Parent extends RequesterResponder {
     /** start worker */
     startWorker(childJsPath: string, terminalError: (e: Error) => any, customArguments: string[]) {
         try {
+            var spawnEnv = process.env;
+            spawnEnv['ATOM_SHELL_INTERNAL_RUN_AS_NODE'] = '1';
```

Comment:
Issues: 
 
a.) You are mutating `process.env` here ... that is bad
b.) Using `process.env` as a whole breaks using *atom as node* on windows. See : https://github.com/atom/atom/issues/2887#issuecomment-93887882 and independently verified : https://github.com/atom/atom/issues/2887#issuecomment-93893894

Solutions :

* Use `Object.create` around process.env
* only do if platform is `NixOS`

Thanks for finding the bug :rose: